libvast/vast/concept/parseable/vast/schema.hpp

```diff
@@ -18,53 +18,183 @@
 #include "vast/concept/parseable/string.hpp"
 #include "vast/concept/parseable/vast/identifier.hpp"
 #include "vast/concept/parseable/vast/type.hpp"
+#include "vast/detail/overload.hpp"
+#include "vast/error.hpp"
+#include "vast/logger.hpp"
 #include "vast/schema.hpp"
 #include "vast/type.hpp"
 
 namespace vast {
 
-struct schema_parser : parser<schema_parser> {
+struct shared_schema_parser : parser<shared_schema_parser> {
+  using symbol_buffer = std::unordered_map<std::string, type>;
   using attribute = schema;
 
-  /// The parser for an identifier.
-  // clang-format off
-  static constexpr auto id
-    = +( parsers::alnum
-       | parsers::ch<'_'>
-       | parsers::ch<'.'>
-       );
-  // clang-format on
+  shared_schema_parser(const symbol_buffer& gs, symbol_buffer& ls)
+    : global_symbols{gs}, local_symbols{ls} {
+    // nop
+  }
 
+  static constexpr auto id = type_parser::id;
   static constexpr auto skp = type_parser::skp;
 
-  template <class Iterator>
-  bool parse(Iterator& f, const Iterator& l, schema& sch) const {
-    type_table symbols;
+  struct resolver {
```

Comment:
Can you document at a high level what this does? It's too generic of a name to understand the purpose from the code.